libs/langchain-google-cloud-sql-pg/src/vectorStore.ts

```diff
@@ -0,0 +1,190 @@
+    let columns: {[key: string]: any} = {};
+
+    for (const index in rows) {
+      const row = rows[index];
+      columns[row["column_name"]] = row["data_type"]
+    }
+
+    if(!columns.hasOwnProperty(idColumn)){
+      throw `Id column: ${idColumn}, does not exist.`;
+    }
+
+    if(!columns.hasOwnProperty(contentColumn)){
+      throw `Content column: ${contentColumn}, does not exist.`;
+    }
+
+    const contentType = columns[contentColumn];
+
+    if(contentType !== "text" && !contentType.includes("char")) {
+      throw `Content column: ${contentColumn}, is type: ${contentType}. It must be a type of character string.`
+    }
+
+    if(!columns.hasOwnProperty(embeddingColumn)) {
+      throw `Embedding column: ${embeddingColumn}, does not exist.`
+    }
+        
+    if(columns[embeddingColumn] !== "USER-DEFINED") {
+      throw `Embedding column: ${embeddingColumn} is not of type Vector.`
+    }
+
+    metadataJsonColumn = columns.hasOwnProperty(metadataJsonColumn) ? metadataJsonColumn : "";
+
+    for (const column of metadataColumns) {
+      if(!columns.hasOwnProperty(column)) {
+        throw `Metadata column: ${column}, does not exist.`
+      }
+    }
+
+    const allColumns = columns;
+    if(ignoreMetadataColumns !== undefined && ignoreMetadataColumns.length > 0) {
+      for (const column of ignoreMetadataColumns) {
+        delete allColumns[column];
+      }
+
+      delete allColumns[idColumn];
+      delete allColumns[contentColumn];
+      delete allColumns[embeddingColumn];
+      metadataColumns = Object.keys(allColumns);
+    } 
+    return new PostgresVectorStore(
+      embeddings,
+      {
+        engine,
+        tableName,
+        schemaName,
+        contentColumn,
+        embeddingColumn,
+        metadataColumns,
+        ignoreMetadataColumns,
+        idColumn,
+        metadataJsonColumn,
+        distanceStrategy,
+        k,
+        fetch_k,
+        lambdaMult,
+        indexQueryOptions
+      }
+    )
+  }
+
+  _vectorstoreType(): string {
+    return "cloudsqlpostgresql"
+  }
+  addVectors(vectors: number[][], documents: DocumentInterface[], options?: { [x: string]: any; }): Promise<string[] | void> {
+    throw new Error("Method not implemented.");
+  }
+  addDocuments(documents: DocumentInterface[], options?: { [x: string]: any; }): Promise<string[] | void> {
+    throw new Error("Method not implemented.");
+  }
+  similaritySearchVectorWithScore(query: number[], k: number, filter?: this["FilterType"] | undefined): Promise<[DocumentInterface, number][]> {
+    throw new Error("Method not implemented.");
+  }
+
+  /**
+   * Deletes documents from the vector store based on the specified parameters.
+   *
+   * @param _params - Flexible key-value pairs defining conditions for document deletion.
+   * @returns A promise that resolves once the deletion is complete.
+   */
+  async delete(_params?: Record<string, any>): Promise<void> { // TODO: test this method
```

Comment:
Let's just support deletion by ids
```suggestion
  async delete(params: {ids?: string[]}): Promise<void> { // TODO: test this method
```